prog/analysis.go

```diff
@@ -297,15 +300,19 @@ func DecodeFallbackSignal(s uint32) (callID, errno int) {
 }
 
 func encodeFallbackSignal(typ, id, aux int) uint32 {
+	checkMaxCallID(id)
 	if typ & ^7 != 0 {
 		panic(fmt.Sprintf("bad fallback signal type %v", typ))
 	}
-	if id & ^fallbackCallMask != 0 {
-		panic(fmt.Sprintf("bad call id in fallback signal %v", id))
-	}
-	return uint32(typ) | uint32(id&fallbackCallMask)<<3 | uint32(aux)<<16
+	return uint32(typ) | uint32(id&fallbackCallMask)<<3 | uint32(aux)<<24
```

Comment:
Can we be sure that 7 bits will always be enough to store aux? Maybe add a panic for this scenario?

Reply:
We put errno there and it's acceptable if they are truncated, we will just get less sensitive feedback signal.
Linux currently has 133 errno's, 8 bits we have are enough to store 256 different values.
Syscalls can yield errno values that are larger than 133, e.g. if syzkaller program installs a seccomp filter that fails all syscalls with large errno value. So if we add a panic, it will fail. But for these fake large errno values truncation is the best way to handle them.